src/wallet/wallet.cpp

```diff
@@ -1818,32 +1818,37 @@ CWallet::ScanResult CWallet::ScanForWalletTransactions(const uint256& start_bloc
     return result;
 }
 
+std::vector<CWalletTx*> CWallet::GetSortedTxs()
+{
+    AssertLockHeld(cs_wallet);
+    // Sort pending transactions based on their initial wallet insertion order
+    std::map<int64_t, CWalletTx*> sorted_txs;
+    for (auto& [wtxid, wtx] : mapWallet) {
+        assert(wtx.GetHash() == wtxid);
```

Comment:
Sorry, I don't understand the need for `assert(wtx.GetHash() == wtxid);`. In which case can this `assert` fails? Only when the data is corrupted?

Reply:
There is never a *need* for an assert. Assertions verify assumptions about the codebase, and should only fail if something you believe to be impossible in the codebase to be the case. It should *never* trigger, unless there is a bug in the code.

By writing this, the code author signals to readers of the code "here you can assume that wtxid is always equal to `wtx.GetHash()`". Further, this is verified during all tests, so we know it is also true for all scenarios any test case runs through.

Reply:
Wait, I see this is just moved and now new, but is this key wtxid or txid? Or does the "w" mean wallet instead of witness?

If it's wtxid, this assertion could fail when wtxid != txid since `CWalletTx::GetHash()` calls `tx->GetHash()`. But AFAICT it is a txid, so should be called txid instead? https://github.com/bitcoin/bitcoin/blob/02aefa169a9e6ed12c7bd8f3392adcd073d8d56b/src/wallet/wallet.cpp#L956-L971

Reply:
The "w" refers to "wallet".

Reply:
Is it good time to rename it to `wallet_txid` for clarification?
Without @glozow comment, i would just assume w stands for witness without any doubt...

Reply:
The variable name also confused me , +1 to renaming it

Reply:
I got the same confusion before reading the entire thread, thought it came from witness...

Reply:
The variable names `wtxid` and `wtx` are used very frequently throughout the wallet code to refer to wallet transactions and their txids. If people find this confusing, then I think it would be more appropriate to have a scripted-diff PR rename all of those rather than renaming each individual instance that occurs in a PR.

Reply:
Renamed to `txid`.